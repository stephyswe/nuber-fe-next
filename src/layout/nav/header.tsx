import clsx from 'clsx';

import { useWindowSizeJs } from '@/hooks/useWindowSizeJs';

import { NavButton } from '@/components/buttons/NavButton';

import { NavProps } from '@/layout/nav/nav';
import { Logo, Spacer } from '@/ui';

type NavHeaderProps = {
  children: JSX.Element;
  change: boolean;
  onSidebar: () => void;
  home: boolean;
} & Pick<NavProps, 'noBorder'>;

export const NavHeader = ({
  children,
  change,
  noBorder,
  onSidebar,
  home,
}: NavHeaderProps) => (
  <header>
    <div className={change ? (home ? 'h-0' : 'h-24') : undefined}>
      <div className={change ? 'fixed top-0 left-0 z-30 w-full ' : undefined}>
        <div className='relative'>
          <div
            className={clsx(
              'right-0 left-0 top-0 m-auto box-border flex h-[96px] items-center justify-between px-4',
              'text-black transition-bg-ease-200 md:min-w-[1024px] md:px-10',
              home ? 'absolute z-10' : 'bg-white',
              change && !noBorder ? 'bg-white box-shadow-rgb-gray' : ''
            )}
          >
            <Navigation onSidebar={onSidebar} change={change} />
            {children}
          </div>
        </div>
      </div>
    </div>
  </header>
);

export function Navigation({
  onSidebar,
  change,
}: Pick<NavHeaderProps, 'onSidebar' | 'change'>) {
  const { isMobile } = useWindowSizeJs();

  if (change && isMobile) return null;

  return (
    <>
      <NavButton onClick={onSidebar} />
      <Spacer className='pr-8 sm:pr-6' />
      <Logo />
      <Spacer className='p-5 md:hidden' />
    </>
  );
}
