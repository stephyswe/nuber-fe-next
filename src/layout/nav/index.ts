export {
  NavbarDiningContent,
  NavbarStoreContent,
  NavCategoryContent,
  NavCityContent,
  NavHomeContent,
} from './content';
export { NavHeader } from './header';
export { NavItemDelivery, NavItemSearch } from './items';
export { Nav } from './nav';
export { NavPerPage } from './perPage';
export { homePosition } from './utils';
