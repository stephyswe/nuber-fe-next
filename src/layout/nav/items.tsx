import clsx from 'clsx';
import { FieldValues, useForm, UseFormRegister } from 'react-hook-form';

import { Input, Link } from '@/components';

import { Spacer } from '@/ui';
import { SvgMap, SvgPosition } from '@/ui/icons';

export const NavItemDelivery = () => (
  <span>
    <Link
      href='/'
      className='box-border flex h-12 cursor-pointer flex-row items-center rounded-[500px] bg-[#eeeeee] px-3 py-2 text-sm font-medium leading-4 text-black hover:bg-[#e2e2e2] active:bg-[#cbcbcb]'
    >
      <div className='h-6 w-4 leading-[1]'>
        <SvgPosition />
      </div>
      <div className='spacer w-2'></div>
      <div className='overflow-hidden text-ellipsis whitespace-nowrap'>
        Map location
      </div>
      <span className='inline-flex items-center text-sm font-medium leading-6 text-[#545454]'>
        &nbsp;&nbsp;•&nbsp;&nbsp;
      </span>
      <div className='whitespace-nowrap'>Pick up now</div>
    </Link>
  </span>
);

export function NavItemSearch({ className, placeholder }: any) {
  const { register } = useForm({
    mode: 'onChange',
  });

  return (
    <div
      role='search'
      aria-label='Sök efter restaurang eller typ av kök'
      className={clsx('flex-grow', className)}
    >
      <div className='relative flex flex-col'>
        <form>
          <label className='absolute m-[-1px] h-[1px] w-[1px] overflow-hidden whitespace-normal clip-rect-1'>
            {placeholder}
          </label>
          <NavItemSearchInput register={register} placeholder={placeholder} />
          <ul className='absolute left-0 right-0 top-full z-10 hidden border-[0_1px_1px_1px] border-solid border-[#e2e2e2] bg-white box-shadow-rgb-10'></ul>
        </form>
      </div>
    </div>
  );
}

const NavItemSearchInput = ({
  register,
  placeholder,
}: {
  register: UseFormRegister<FieldValues>;
  placeholder: string;
}) => (
  <div className='relative box-border flex min-w-full border-none bg-[#eeeeee] px-4 py-2 text-base font-normal leading-6 text-black transition-bs-ease-300 box-shadow-inset-eee focus-within:box-shadow-rgb-0'>
    <div className='flex flex-shrink-0 items-center py-2'>
      <div className='h-6 w-6 leading-[1]'>
        <SvgMap />
      </div>
    </div>
    <Spacer className='w-4' />
    <Input
      variant='input3'
      type='text'
      role='combobox'
      name='searchTerm'
      placeholder={placeholder}
      register={register}
    />
    <Spacer className='w-2' />
  </div>
);
