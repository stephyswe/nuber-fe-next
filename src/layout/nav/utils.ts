export function homePosition(
  position: any,
  setChange: (change: boolean) => void,
  change: boolean,
  changePos: number,
  changeMob: number,
  isMobile: boolean
) {
  if (position && !isMobile) {
    if (position.scrollY > changePos && !change) setChange(true);
    if (position.scrollY <= changePos && change) setChange(false);
  }

  if (position && isMobile) {
    if (position.scrollY > changeMob && !change) setChange(true);
    if (position.scrollY <= changeMob && change) setChange(false);
  }
}
