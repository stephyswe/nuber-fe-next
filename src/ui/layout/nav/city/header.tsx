import clsx from 'clsx';

import { NavButton } from '@/components/buttons/NavButton';

import { NewLogo, Spacer } from '@/ui';

export function NavCityHeader({
  children,
  change,
  noHoverBorder,
  onClickSidebar,
}: any) {
  return (
    <header>
      <div className={change ? 'h-24' : undefined}>
        <div className={change ? 'fixed top-0 left-0 z-30 w-full ' : undefined}>
          <div className='relative'>
            <div
              className={clsx(
                'right-0 left-0 top-0 m-auto box-border flex h-[96px] items-center justify-between px-4 text-black transition-bg-ease-200 md:min-w-[1024px] md:px-10',
                change && !noHoverBorder ? 'box-shadow-rgb-gray' : '',
                'bg-white'
              )}
            >
              <Navigate onClickSidebar={onClickSidebar} />
              {children}
            </div>
          </div>
        </div>
      </div>
    </header>
  );
}

const Navigate = ({ onClickSidebar }: any) => (
  <>
    <NavButton onClick={onClickSidebar} />
    <Spacer className='w-8' />
    <NewLogo />
  </>
);
