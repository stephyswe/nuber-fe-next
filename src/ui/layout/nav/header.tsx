import clsx from 'clsx';

import { useWindowSizeJs } from '@/hooks/useWindowSizeJs';

import { NavButton } from '@/components/buttons/NavButton';

import { NewLogo, Spacer } from '@/ui';

type NavHeaderProps = {
  children: JSX.Element;
  change: boolean;
  noHoverBorder: boolean;
  onSidebar: () => void;
  home: boolean;
};

export const NavHeader = ({
  children,
  change,
  noHoverBorder,
  onSidebar,
  home,
}: NavHeaderProps) => (
  <header>
    <div className={change ? (home ? 'h-0' : 'h-24') : undefined}>
      <div className={change ? 'fixed top-0 left-0 z-30 w-full ' : undefined}>
        <div className='relative'>
          <div
            className={clsx(
              'right-0 left-0 top-0 m-auto box-border flex h-[96px] items-center justify-between px-4',
              'text-black transition-bg-ease-200 md:min-w-[1024px] md:px-10',
              home ? 'absolute z-10' : 'bg-white',
              change && !noHoverBorder ? 'bg-white box-shadow-rgb-gray' : ''
            )}
          >
            <Navigation onSidebar={onSidebar} change={change} />
            {children}
          </div>
        </div>
      </div>
    </div>
  </header>
);

export function Navigation({
  onSidebar,
  change,
}: Pick<NavHeaderProps, 'onSidebar' | 'change'>) {
  const { isMobile } = useWindowSizeJs();

  if (isMobile) {
    return (
      <>
        {!change ? (
          <>
            <NavButton onClick={onSidebar} />
            <Spacer className='pr-6' />
            <NewLogo />
            <Spacer className='p-5' />
          </>
        ) : null}
      </>
    );
  } else {
    return (
      <>
        <NavButton onClick={onSidebar} />
        <Spacer className='pr-8' />
        <NewLogo />
      </>
    );
  }
}
