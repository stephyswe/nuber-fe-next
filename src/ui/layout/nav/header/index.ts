export { NavClientHeader } from './client';
export { NavHomeHeader } from './home';
