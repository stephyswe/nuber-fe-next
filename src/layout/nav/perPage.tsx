import { useRouter } from 'next/router';

import { NavCategoryContent } from '@/layout/nav/content/client/category';
import { NavCityContent } from '@/layout/nav/content/client/city';
import { NavbarDiningContent } from '@/layout/nav/content/client/dining';
import { NavbarStoreContent } from '@/layout/nav/content/client/store';
import { NavHomeContent } from '@/layout/nav/content/home';

type NavPerPageProps = {
  change: boolean;
  onScroll: boolean;
};

export const NavPerPage = (props: NavPerPageProps) => {
  const { pathname } = useRouter();
  switch (pathname) {
    case '/':
      return <NavHomeContent {...props} />;

    case '/client/city/[label]':
      return <NavCityContent {...props} />;

    case '/client/category/[label]/[sublabel]':
      return <NavCategoryContent {...props} />;

    case '/client/region/[label]':
      return <NavCityContent />;

    case '/client/dining/pickup':
      return <NavbarDiningContent />;

    case '/client/dining/delivery':
      return <NavbarDiningContent />;

    case '/client/store/[label]':
      return <NavbarStoreContent />;

    default:
      break;
  }

  return <div>no case</div>;
};
