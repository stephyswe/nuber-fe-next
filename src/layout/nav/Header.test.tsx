import { render, renderMobile, screen, takeSnapshot } from 'test-utils';

import { NavHeader } from '@/layout/nav/header';

function RenderHeader(props: any) {
  return (
    <NavHeader onSidebar={jest.fn()} change={false} home={false} {...props} />
  );
}

// ** Snapshot ** //
takeSnapshot(<RenderHeader />);

// ** Desktop Size ** //

test('has nav button component', () => {
  render(<RenderHeader />);
  const navBtn = screen.getByRole('button', { name: 'nav-btn' });
  expect(navBtn).toBeInTheDocument();
});

test('has link component', () => {
  render(<RenderHeader />);
  const link = screen.getByRole('link', { name: 'uber-home-link' });
  expect(link).toBeInTheDocument();
});

test('has logo component', () => {
  render(<RenderHeader />);
  const logo = screen.getByRole('img', { name: 'uber-logo' });
  expect(logo).toBeInTheDocument();
});

// ** Mobile Size ** //

// ** Mobile Size ** - Change True //

test('on mobile - has no logo component', () => {
  renderMobile(<RenderHeader change={true} />);
  expect(screen.queryByTestId('ui-logo')).not.toBeInTheDocument();
});

test('on mobile - has content', () => {
  const { container } = renderMobile(
    <RenderHeader home={true} change={true} />
  );
  expect(container).toContainHTML(
    '<div><header><div class="h-0"><div class="fixed top-0 left-0 z-30 w-full "><div class="relative"><div class="right-0 left-0 top-0 m-auto box-border flex h-[96px] items-center justify-between px-4 text-black transition-bg-ease-200 md:min-w-[1024px] md:px-10 absolute z-10 bg-white box-shadow-rgb-gray" /></div></div></div></header></div>'
  );
});

// ** Desktop Size - Spacer conditional ** //

test('on desktop - has two spacer item', () => {
  render(<RenderHeader />);
  const spacer = screen.getAllByTestId('ui-spacer');
  expect(spacer).toHaveLength(2);
});

// ** Mobile Size - Spacer conditional ** //

// TODO:
