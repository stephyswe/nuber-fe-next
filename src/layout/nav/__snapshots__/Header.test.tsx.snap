// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header>
  <div>
    <div>
      <div
        className="relative"
      >
        <div
          className="right-0 left-0 top-0 m-auto box-border flex h-[96px] items-center justify-between px-4 text-black transition-bg-ease-200 md:min-w-[1024px] md:px-10 bg-white"
        >
          <button
            aria-label="nav-btn"
            className="flex h-6 w-6 cursor-pointer items-center"
            onClick={[MockFunction]}
          >
            <svg
              aria-hidden="true"
              className="icon_20"
              focusable="false"
              viewBox="0 0 20 20"
            >
              <path
                d="M19.167 3.333H.833v2.5h18.334v-2.5zm0 5.834H.833v2.5h18.334v-2.5zM.833 15h18.334v2.5H.833V15z"
              />
            </svg>
          </button>
          <div
            className="m-0 flex-shrink-0 h-[1px] pr-8 sm:pr-6"
            data-testid="ui-spacer"
          />
          <a
            aria-label="uber-home-link"
            className=""
            data-testid="component-link"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <img
              alt="uber-logo"
              className="block h-6 max-h-[18px] w-auto max-w-[136px] md:max-h-[inherit] md:max-w-[inherit] "
              height="24"
              src="/images/logo.svg"
              width="146"
            />
          </a>
          <div
            className="m-0 flex-shrink-0 h-[1px] p-5 md:hidden"
            data-testid="ui-spacer"
          />
        </div>
      </div>
    </div>
  </div>
</header>
`;
